caas/kubernetes/provider/bootstrap.go

```diff
@@ -343,24 +345,30 @@ func (c *controllerStack) Deploy() (err error) {
 	return nil
 }
 
-func (c *controllerStack) getControllerSvcSpec(cloudType string) (*controllerServiceSpec, error) {
+func (c *controllerStack) getControllerSvcSpec(cloudType string, cfg *podcfg.BootstrapConfig) (*controllerServiceSpec, error) {
 	spec, ok := controllerServiceSpecs[cloudType]
 	if !ok {
 		logger.Debugf("fallback to default svc spec for %q", cloudType)
 		spec = controllerServiceSpecs[caas.K8sCloudOther]
 	}
-	if spec == nil || spec.ServiceType == "" {
+	if cfg != nil && cfg.ControllerServiceType != "" {
+		spec.ServiceType = core.ServiceType(cfg.ControllerServiceType)
```

Comment:
This is wrong. See https://github.com/juju/juju/blob/57deffaf61559ec3ff6ceeb31f2d3ef1e22e3999/caas/kubernetes/provider/k8s.go#L1650, it needs to be converted from a juju service type into a k8s service type.

Reply:
yeah, good catch, fixed